.github/workflows/onpush-dotnet-test-and-create-release.yml

```diff
@@ -0,0 +1,31 @@
+name: Unit test and create new release
```

Comment:
Need to think about filenames here. This is no longer an `onpush` action, so feels strange to keep calling it this. Maybe just needs to be `dotnet_test-and-create-release-version.yml`?

If we are going to have multiple shared workflows in here for different project types, maybe the naming convention should include the technology stack as the prefix, e.g `dotnet_`?

Reply:
The rationale for leaving onpush/onrelease in the names is that's where it will generally be called from - and yes, I did include `dotnet` in the name 😄

Reply:
Files renamed